data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeCreatorImpl.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.knative.eventing.kafka.broker.core.eventtype;
+
+import dev.knative.eventing.kafka.broker.contract.DataPlaneContract;
+import io.cloudevents.CloudEvent;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.OwnerReferenceBuilder;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.informers.cache.Lister;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.WorkerExecutor;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventTypeCreatorImpl implements EventTypeCreator {
+
+    private static final Integer DNS1123_SUBDOMAIN_MAX_LENGTH = 253;
+
+    private final MixedOperation<EventType, KubernetesResourceList<EventType>, Resource<EventType>> eventTypeClient;
+
+    private final Lister<EventType> eventTypeLister;
+
+    private MessageDigest messageDigest;
+
+    private final WorkerExecutor executor;
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeCreatorImpl.class);
+
+    public EventTypeCreatorImpl(
+            MixedOperation<EventType, KubernetesResourceList<EventType>, Resource<EventType>> eventTypeClient,
+            Lister<EventType> eventTypeLister,
+            Vertx vertx) {
+        this.eventTypeClient = eventTypeClient;
+        this.eventTypeLister = eventTypeLister;
+        this.executor = vertx.createSharedWorkerExecutor("et-creator-worker", 1);
+        try {
+            this.messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ignored) {
+            this.messageDigest = null;
+        }
```

Comment:
Catching and leaving it to null, will just delay the exception to the runtime, can we just not catch it or if it's unchecked exception, we could wrap it into a runtime exception?

Reply:
Currently it doesn't throw a runtime exception, the autocreate just becomes a no-op (no eventtype gets created). Do you think it's better to fail here instead?